ui/desktop/src/components/ApiKeyTester.tsx

```diff
@@ -0,0 +1,232 @@
+
+        onSuccess(provider_name, models[0]);
+      }
+    } catch (error: unknown) {
+      setTestResults([
+        {
+          provider: 'Unknown',
+          success: false,
+          error: 'Could not detect provider from API key',
+          suggestions: [
+            'Make sure you are using a valid API key from OpenAI, Anthropic, Google, Groq, or xAI',
+            'Check that the key is complete and not truncated',
+            'Verify your API key is active and has sufficient credits',
+            'For local Ollama setup, use the "Other Providers" section below',
+          ],
+        },
+      ]);
+
+      toastService.error({
+        title: 'Detection Failed',
+        msg: 'Could not detect provider. Please check your API key and try again.',
+        traceback: error instanceof Error ? error.stack || '' : '',
+      });
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const hasInput = apiKey.trim().length > 0;
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setApiKey(e.target.value);
+  };
+
+  return (
+    <div className="relative w-full mb-6">
+      {/* Recommended pill */}
+      <div className="absolute -top-2 -right-2 sm:-top-3 sm:-right-3 z-20">
+        <span className="inline-block px-2 py-1 text-xs font-medium bg-blue-600 text-white rounded-full">
+          Recommended
+        </span>
+      </div>
+
+      <div className="w-full p-4 sm:p-6 bg-background-muted border border-background-hover rounded-xl">
+        <div className="flex items-start justify-between mb-3">
+          <div className="flex-1">
+            <Key className="w-4 h-4 mb-3 text-text-standard" />
+            <h3 className="font-medium text-text-standard text-sm sm:text-base">
+              Quick Setup with API Key
+            </h3>
+          </div>
+        </div>
+
+        <p className="text-text-muted text-sm sm:text-base mb-4">
+          Enter your API key and we'll automatically detect which provider it works with.
+        </p>
+
+        <div className="space-y-4">
+          <div className="flex gap-2 items-stretch">
+            <input
+              ref={inputRef}
+              type="password"
+              value={apiKey}
+              onChange={handleInputChange}
+              placeholder="Enter your API key (OpenAI, Anthropic, Google, etc.)"
+              className="flex-1 px-3 py-2 border border-background-hover rounded-lg bg-background-default text-text-standard placeholder-text-muted focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"
+              disabled={isLoading}
+              onKeyDown={(e) => {
+                if (e.key === 'Enter' && !isLoading && hasInput) {
+                  testApiKey();
+                }
+              }}
+            />
+            <Button
+              onClick={testApiKey}
+              disabled={isLoading || !hasInput}
+              variant={hasInput && !isLoading ? 'default' : 'secondary'}
```

Comment:
disabled and variant follow the same logic differently formulated, let's make that easier to read